File: 940-fruit-into-baskets/fruit-into-baskets.java
import java.util.*;
class Solution {
    public int totalFruit(int[] fruits) {
        int low=0, result=Integer.MIN_VALUE;
        HashMap<Integer, Integer> hashmap = new HashMap<>();
        for(int high=0; high<fruits.length; high++)
        {
            hashmap.put(fruits[high], hashmap.getOrDefault(fruits[high], 0) + 1);

            while(hashmap.size() > 2){
                hashmap.put(fruits[low], hashmap.get(fruits[low]) - 1);
                if(hashmap.get(fruits[low]) == 0){
                    hashmap.remove(fruits[low]);
                }
                low++;
            }
            int length = high - low + 1;
            result = Math.max(result, length);
        }
        return result;
    }
}

File: 940-fruit-into-baskets/fruit-into-baskets.py
class Solution:
    def totalFruit(self, fruits: list[int]) -> int:
        high=0
        low=0
        res=float("-inf")
        freq={}
        n=len(fruits)
        while(high<n):
            if fruits[high] in freq:
                freq[fruits[high]] = freq[fruits[high]] + 1
            else:
                freq[fruits[high]] = 1

            while len(freq) > 2:
                freq[fruits[low]] = freq[fruits[low]] - 1
                if freq[fruits[low]] == 0:
                    del freq[fruits[low]]
                
                low+=1
            
            window = high - low + 1
            res = max(res, window)

            high+=1
        return res